backend/de.metas.business/src/main/java/de/metas/forex/ForexContractService.java

```diff
@@ -104,20 +108,18 @@ public void allocateOrder(
 				});
 	}
 
+	@NonNull
 	public BooleanWithReason checkOrderEligibleToAllocate(final OrderId orderId)
 	{
-		final I_C_Order order = orderBL.getById(orderId);
-		return checkOrderEligibleToAllocate(order);
+		return checkOrderCompleted(orderId);
 	}
 
-	private static BooleanWithReason checkOrderEligibleToAllocate(final I_C_Order order)
+	@NonNull
+	private BooleanWithReason checkOrderCompleted(@NonNull final OrderId orderId)
```

Comment:
IMHO you should keep both variants of the method, 
* with I_C_Order param (as it was)
* with OrderId param => load the order and call the other method

because it's not efficient if you have the order loaded to ignore that fact, call methods using orderId and then load the order again